ogre2/src/Ogre2ThermalCamera.cc

```diff
@@ -353,6 +368,43 @@ void Ogre2ThermalCameraMaterialSwitcher::preRenderTargetUpdate(
           subItem->setMaterial(this->heatSignatureMaterials[item->getId()]);
         }
       }
+      // background objects
+      else
+      {
+        Ogre::Aabb aabb = item->getWorldAabbUpdated();
+        Ogre::AxisAlignedBox box = Ogre::AxisAlignedBox(aabb.getMinimum(), aabb.getMaximum());
+
+        // we will be converting rgb values to tempearture values in shaders
+        // but we want to make sure the object rgb values are not affected by lighting
+        // so disable lighting
+        // Also check if objects are within camera view
+        if (ogreVisual->GeometryCount() > 0u && this->ogreCamera->isVisible(box))
+        {
+          auto geom = ogreVisual->GeometryByIndex(0);
```

Comment:
attached geometries always start from index 0. Since there is a check above for `ogreVisual->GeometryCount() > 0`, there should be a valid geometry here. I can also add an extra check for null ptr to be sure.

Reply:
added check 397a443